balancer-js/src/modules/zaps/bbausd2-migrations/bbausd1.ts

```diff
@@ -0,0 +1,312 @@
+import { ADDRESSES } from './addresses';
+import { Relayer } from '@/modules/relayer/relayer.module';
+import { BatchSwapStep, FundManagement, SwapType } from '@/modules/swaps/types';
+import { Interface } from '@ethersproject/abi';
+import { MaxUint256, Zero } from '@ethersproject/constants';
+// TODO - Ask Nico to update Typechain?
+import balancerRelayerAbi from '@/lib/abi/BalancerRelayer.json';
+import { BigNumber } from '@ethersproject/bignumber';
+const balancerRelayerInterface = new Interface(balancerRelayerAbi);
+
+const SWAP_RESULT_BBAUSD = Relayer.toChainedReference('24');
+
+export class BbaUsd1Builder {
+  private addresses;
+
+  constructor(networkId: 1 | 5) {
+    this.addresses = ADDRESSES[networkId];
+  }
+
+  /**
+   * Builds migration call data.
+   * Migrates tokens from bbausd1 to bbausd2 pool.
+   * Tokens that are initially staked are re-staked at the end of migration. Non-staked are not.
+   *
+   * @param {string}    userAddress User address.
+   * @param {string}    bbausd1Amount Amount of BPT tokens to migrate.
+   * @param {string}    minBbausd2Out Minimum of expected BPT out ot the migration flow.
+   * @param {boolean}   staked Indicates whether tokens are initially staked or not.
+   * @param {string[]}  tokenBalances Token balances in EVM scale. Array must have the same length and order as tokens in pool being migrated from. Refer to [getPoolTokens](https://github.com/balancer-labs/balancer-v2-monorepo/blob/master/pkg/interfaces/contracts/vault/IVault.sol#L334).
+   * @param {string}    authorisation Encoded authorisation call.
+   * @returns Migration transaction request ready to send with signer.sendTransaction
+   */
+  calldata(
+    userAddress: string,
+    bbausd1Amount: string,
+    minBbausd2Out: string,
+    staked: boolean,
+    tokenBalances: string[],
+    authorisation?: string
+  ): {
+    to: string;
+    data: string;
+  } {
+    const relayer = this.addresses.relayer;
+    let calls: string[] = authorisation
+      ? [this.buildSetRelayerApproval(authorisation)]
```

Comment:
i'm in favour of consistently using `if` across the fn body here, easier to read

Reply:
✅